src/odemis/driver/autoscript_client.py

```diff
@@ -0,0 +1,2065 @@
+            self._generator = None
+
+    def start_generate(self) -> None:
+        self._genmsg.put(GEN_START)
+        if not self._generator or not self._generator.is_alive():
+            logging.info("Starting acquisition thread")
+            self._generator = threading.Thread(target=self._acquire,
+                                               name="autoscript acquisition thread")
+            self._generator.start()
+
+    def stop_generate(self) -> None:
+        self.stop_acquisition(wait_for_frame=False)
+        self._genmsg.put(GEN_STOP)
+        # TODO: add a cancel once scanning is asynchronous
+
+    def _acquire(self) -> None:
+        """
+        Acquisition thread
+        Managed via the ._genmsg Queue
+        """
+        try:
+            while True:
+                # Wait until we have a start (or terminate) message
+                self._acq_wait_start()
+                logging.debug("Preparing acquisition")
+                while True:
+                    logging.debug("Start acquiring an image")
+
+                    # HACK: from xt_client to prevent double scanning
+                    if self._acq_should_stop(timeout=0.0):
+                        logging.debug("Image acquisition should stop, exiting loop")
+                        break
+
+                    # NOTE: the stage metadata is late when acquiring an overview (it is for the previous tile)
+                    # we need to do some kind of synchronization to get the right stage position
+                    time.sleep(0.2) # TODO: do something more intelligent
+
+                    md = self._scanner._metadata.copy()
+                    if hasattr(self._scanner, "dwellTime"):
+                        md[model.MD_BEAM_DWELL_TIME] = self._scanner.dwellTime.value
+                    if hasattr(self._scanner, "rotation"):
+                        md[model.MD_BEAM_SCAN_ROTATION] = self._scanner.rotation.value
+                    if hasattr(self._scanner, "accelVoltage"):
+                        md[model.MD_BEAM_VOLTAGE] = self._scanner.accelVoltage.value
+                    if hasattr(self._scanner, "beamCurrent"):
+                        md[model.MD_BEAM_CURRENT] = self._scanner.beamCurrent.value
+                    if hasattr(self._scanner, "shift"):
+                        md[model.MD_BEAM_SHIFT] = self._scanner.shift.value
+                    if hasattr(self._scanner, "horizontalFoV"):
+                        md[model.MD_BEAM_FIELD_OF_VIEW] = self._scanner.horizontalFoV.value
+                    if hasattr(self._scanner, "_acq_type"):
+                        md[model.MD_ACQ_TYPE] = self._scanner._acq_type
+                    md[model.MD_ACQ_DATE] = time.time()
+                    md.update(self._metadata)
+
+                    # Estimated time for an acquisition is the dwell time times the total amount of pixels in the image.
+                    if hasattr(self._scanner, "dwellTime") and hasattr(self._scanner, "resolution"):
```

Comment:
Are there cases where these VAs are missing? I feel it's just some left-over from the xt_client which has a very basic FIB scanner without dwellTime.